clab/config/utils.go

```diff
@@ -13,7 +12,16 @@ import (
 )
 
 const (
```

Comment:
This is a short summary of the keys, that are optionally supplied and in the resulting vars sent to the template

Reply:
why not define these as functions that can be used within the template ?

Reply:
node / nodes / links / role / far really are mechanisms to expose the topology to the template, they have no alternative and should generally not be used in a node's `vars:` (role is probably the exception for more advanced use-caese)

ip / name / linknr are convenience `vars` to render templates, These are obviously optional, but used in most networks (except if you go unnumbered). It would also be really cool to visualize / `graph` them.

Note: The second set can be templated, but it would increase complexity a lot for this first drop.

Also, generators (like IP pools, unique numbers) will probably end up in the same place (while preparing vars) They might not have fixed names, but the point is that I dont see it as part of the template, as they would need to keep state across templates and even across different executions and are used by different templates.

Reply:
ok probably not in this PR